completed enumerable method

vidhishweta01 opened the pull request:
Enumerables_assignment is a project which contain the implementation of original method behaviour in user defined function
such as all? to my_all? 
it contains various methods which behave same as original method please give your valuable feedback

vidhishweta01 committed:
completed enumerable method

vidhishweta01 commented:
Hii, I have finished all the methods of Enumerable using yield and self kindly give your valuable feedback

mke2111 commented:
> Hii, I have finished all the methods of Enumerable using yield and self kindly give your valuable feedback

I'm looking through your project. Let's have a chat on slack.

vidhishweta01 committed:
added readme file

vidhishweta01 committed:
edited github workflow

vidhishweta01 committed:
edited github workflow

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fix linter errors

vidhishweta01 committed:
fix linter errors

mke2111 reviewed (changes requested):
# **STATUS: CHANGES REQUIRED 🔴** 

Hey @vidhishweta01, Thanks for the great work done on the project. Your enumerable methods work just fine except for a few test cases.

### REQUIRED CHANGES 🔴 

- [ ] In your subsequent projects from this point, kindly have a number of commits. They serve a great purpose especially when you want to roll back to a certain point in your project. Also, make them descriptive for everyone to understand.
- [ ] Add a readme file for your Project. This should be descriptive in all sections. Show how to get started with your methods on a local computer. Also, include the authors' details.


```
array = [6, 19, 19, 25, 7, 30, 20, 27, 22, 19, 18, 29, 12, 31, 2, 12, 0, 32, 1, 20]
block = proc { |num| num < (0 + 9) / 2 }
rang = Range.new(5, 50)
false_block = proc { |num| num > 9 }
true_block = proc { |num| num <= 9 }
numbers = [1, 2i, 3.14]
words = %w[programmer computer house car]
true_any_array = [nil, false, true, []]
false_any_array = [nil, false, nil, false]
true_array = [1, true, 'hi', []]
false_array = [1, false, 'hi', []]
```

- Use these for your test cases.


- [ ] ` my_each ` when a block is given when self is an array returns the array itself after calling the given block once for each element in self. It is instead returning the count.
- [ ] ` my_each ` when a block is given when self is a range returns the range itself after calling the given block once for each element in self.
- [ ] ` my_each ` when a block is given when self is a hash should behave like ` each ` and calls the given block once for each element in self. Your method returns an empty string.

- [ ] ` my_select ` returns an array containing all elements of enum for which the given block returns a true value. `my_select ` doesn't behave like the `select` enumerable.
- [ ] ` my_select ` should return an array containing all elements of enum for which the given block returns a true value::range.

- [ ] ` my_all? ` shoud return false if the block returns false or nil::range
- [ ] ` my_all? ` when no block or argument is given returns false when one of the collection members are false or nil. expecting ` false ` but your method returns ` true `.

- [ ] `my_none? ` when no block or argument is given returns true only if none of the collection members is true. expecting ` true ` but your method returns ` false `.
- [ ] ` my_none? ` when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern

- [ ] `my_count` returns the number of items in enum through enumeration::range.
- [ ] `my_count` counts the number of elements yielding a true value if a block is given.

- [ ] `my_map` returns a new array with the results of running a given block once for every element in enum::range

- [ ] `my_inject` raises a "LocalJumpError" when no block or argument is given.
- [ ] `my_inject` when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method. Your method only accepts arguments as a range.
- [ ] `my_inject` when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method::range.  Your method only accepts arguments as a range.

- [ ] ` multiply_els ` accepts an array as an argument and multiplies all the elements of the array together using ` my_inject `. Returns `nil`.


### Kindly make all the requested changes 💯  and submit another CRR. 👯 

Happy Coding! 😄 

Reach me on slack for a ChitChat 🆘 : **Roy Mukuye**

vidhishweta01 committed:
changes done as per code review

vidhishweta01 committed:
changes done as per code review

vidhishweta01 committed:
removed linter error

vidhishweta01 commented:
Hii, I have changed as per code review:
1. my_each
2. my_none
3. my_all
4. my_inject
5. my_map
6. more explanation in readme file

descholar-ceo commented:
# This Pull Request is under review by [descholar-ceo](https://github.com/descholar-ceo) :eyes: :eyes: 
## Please wait a bit ...

descholar-ceo commented on test5.rb:
```diff
@@ -0,0 +1,136 @@
+    result
+  end
+
+  def my_all?(pattern = nil)
+    c = 0
+    statement = true
+    if block_given?
+      to_a.length.times do
+        statement = false unless yield(to_a[c])
+        c += 1
+      end
+    elsif !pattern.nil?
+      to_a.length.times do
+        statement = false if to_a[i].scan(pattern)
+      end
+    end
+    statement
+  end
```
### #my_all? should behave as follow:
- [ ]  1. when no block or argument is given returns false when one of the collection members are false or nil
- [ ] 2. when a class is passed as an argument returns true if all of the collection is a member of such class::Integer
- [ ] 3. when a class is passed as an argument returns true if all of the collection is a member of such class::Numeric
- [ ] 4. when a class is passed as an argument returns false if any of the collection is not a member of such class
- [ ] 5. when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [ ] 6. when a Regex is passed as an argument returns false if any of the collection does not match the Regex
- [ ] 7. when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
- [ ] 8. when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern

descholar-ceo commented on test5.rb:
```diff
@@ -0,0 +1,136 @@
+    statement
+  end
+
+  def my_any?(pattern = nil)
+    c = 0
+    statement = false
+    if block_given?
+      length.times do
+        statement = true if yield(self[c])
+        c += 1
+      end
+    elsif !pattern.nil?
+      length.times do
+        statement = true if self[i].scan(pattern)
+      end
+    end
+    statement
+  end
```
### my_any? should behave as follow:
- [ ] 9. returns false if the block ever returns a value false or nil::range
- [ ] 10. when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] 11. when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric
- [ ] 12. when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer
- [ ] 13. when a Regex is passed as an argument returns true if any of the collection matches the Regex
- [ ] 14. when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] 15. when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
- [ ] 16. when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern

descholar-ceo commented on test5.rb:
```diff
@@ -0,0 +1,136 @@
+    statement
+  end
+
+  def my_none?(pattern = nil)
+    c = 0
+    statement = true
+    if block_given?
+      to_a.length.times do
+        statement = false if yield(to_a[c])
+        c += 1
+      end
+    elsif !pattern.nil?
+      to_a.length.times do
+        statement = false if to_a[i].scan(pattern)
+      end
+    end
+    statement
+  end
```
### #my_none? should behave as follow:
- [ ] 17. when no block or argument is given returns false only if one of the collection members is true
- [ ] 18. when a class is passed as an argument returns true if none of the collection is a member of such class
- [ ] 19. when a class is passed as an argument returns true if none of the collection is a member of such class::Numeric
- [ ] 20. when a class is passed as an argument returns false if any of the collection is a member of such class
- [ ] 21. when a Regex is passed as an argument returns true if none of the collection matches the Regex
- [ ] 22. when a Regex is passed as an argument returns false if any of the collection matches the Regex
- [ ] 23. when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
- [ ] 24. when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern

descholar-ceo commented on test5.rb:
```diff
@@ -0,0 +1,136 @@
+    statement
+  end
+
+  def my_count(var = nil)
+    c = 0
+    if !var.nil?
+      to_a.length.times do |i|
+        c += 1 if to_a[i] == var
+      end
+      c
+    else
+      to_a.length
+    end
+  end
```
### #my_count should behave as follow:
- [ ] 25. counts the number of elements yielding a true value if a block is given

descholar-ceo reviewed (changes requested):
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @vidhishweta01 :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your peoject is very good, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implemented the requested changes from the previous review, keep it up, and your project is really almost done.
#### :warning: However we need to make few changes to mark this project approved, don't feel discouraged because of being asked some changes more and more, that's beacuse this is a learning process, and we are here to help you to become an out-standing developer. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [ ] Add description to this pull request please
- [ ] I have written down all of the failing test cases for all the enumerables, please check them out from inline comments :point_down: 
___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

vidhishweta01 committed:
corrected my_all, my_none, my_any as per the code review

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
fixed linter errors

vidhishweta01 committed:
edited count

vidhishweta01 committed:
edited count

vidhishweta01 commented:
Hii, I have changed as per code review:

my_all?
my_none?
my_any?
my_count
kindly give your valuable feedback

rindrajosia reviewed (changes requested):
**Changes required** :red_circle: 
Hello,

Well done job so far. However, there are bugs to fix.

- [ ] [REQUIRED] #my_all? when no block or argument is given returns false when one of the collection members are false or nil
- [ ] [REQUIRED] #my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [ ] [REQUIRED] #my_all? when a Regex is passed as an argument returns false if any of the collection does not match the Regex
- [ ] [REQUIRED] #my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
- [ ] [REQUIRED] #my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern
- [ ] [REQUIRED] #my_any? returns false if the block ever returns a value false or nil::range
- [ ] [REQUIRED] #my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] [REQUIRED] #my_any? when a Regex is passed as an argument returns true if any of the collection matches the Regex
- [ ] [REQUIRED] #my_any? when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] [REQUIRED] #my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
- [ ] [REQUIRED] #my_any? when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern
- [ ] [REQUIRED] #my_none? when no block or argument is given returns false only if one of the collection members is true
- [ ] [REQUIRED] #my_none? when a Regex is passed as an argument returns true if none of the collection matches the Regex
- [ ] [REQUIRED] #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
- [ ] [REQUIRED] #my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern
- [ ] [REQUIRED] #my_count counts the number of elements yielding a true value if a block is given


Kindly make the changes and submit a new code review request.
Thanks
Rindra Josia

vidhishweta01 committed:
corrected my_all, my_none, my_any as per the code review

vidhishweta01 commented:
Hii, I have changed as per code review:

my_all?
my_none?
my_any?
I have added the description in readme for these how they work with regular expression

mke2111 reviewed (changes requested):
# **STATUS: CHANGES REQUIRED 🔴 ⚠️** 

## Hey @vidhishweta01, 👋🏽 
 **Thank you for the good job you got going on. The enumerable methods run just fine and almost pass all test cases.**

## REQUIRED CHANGES

- [ ] Kindly fix all the methods listed by the previous reviewer. There are 14 tests failing from `my_all?`, `my_any?`, `my_none?`, `my_count`. 
- [ ] Kindly write a descriptive readme file as well.

**Kindly make all the requested changes and submit another CRR. 👍🏽** 

## Cheers!! 🍹 & Happy Coding.

# Reach me on slack for a chitchat 🆘  **Roy Mukuye**

vidhishweta01 committed:
changes made as per code review

vidhishweta01 commented:
Hi I have made changes in following as per given recommendation and checked 
my_all
my_none
my_any
my_count
and added the tests for each in readme file
please share your valuable feedback

umairarshadbutt reviewed (changes requested):
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @vidhishweta01 
Good job with the changes! :tada: We are almost there.
We, however, do have a few changes to carry out before we proceed (see review comment).

##

Kindly work on the suggested changes 👇.
- [ ] Kindly fix all the methods listed by the previous reviewer. There are 8 tests failing from `my_all?`, `my_any?`, `my_none?`.
##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

vidhishweta01 committed:
changes made as per code review

vidhishweta01 commented:
Hi I have made changes in following as per given recommendation and checked
my_all
my_none
my_any
now they are working for nil, false and true pattern as given in true_array, any_true_array, false_array as original methods
please share your valuable feedback

rindrajosia reviewed (changes requested):
**Required Changes**
Hi, I'm Josia your reviewer for this PR 
Good job improving your project! There are still some issues to revise before approving this project but hang in there and read the comments below with care. You can do it! :muscle: 

- [ ] [REQUIRED] #my_all? returns true if the block never returns false or nil
- [ ] [REQUIRED] #my_all? returns false if the block returns false or nil
- [ ] [REQUIRED] #my_all? returns false if the block returns false or nil::range
- [ ] [REQUIRED] #my_all? when no block or argument is given returns true when none of the collection members are false or nil
- [ ] [REQUIRED] #my_all? when a Regex is passed as an argument returns false if any of the collection does not match the Regex
- [ ] [REQUIRED] #my_any? returns false if the block ever returns a value false or nil
- [ ] [REQUIRED] #my_any? returns false if the block ever returns a value false or nil::range
- [ ] [REQUIRED] #my_any? when no block or argument is given returns false if at least one of the collection is not true
- [ ] [REQUIRED] #my_none? when no block or argument is given returns false only if one of the collection members is true

_**This project is almost done!**_
Happy coding
Rindra Josia

vidhishweta01 committed:
changes made as per code review

vidhishweta01 committed:
changes made as per code review

vidhishweta01 committed:
changes made as per code review

vidhishweta01 commented:
Worked on these changes as per the code review
 #my_all? returns true if the block never returns false or nil
  #my_all? returns false if the block returns false or nil
  #my_all? returns false if the block returns false or nil::range
 #my_all? when no block or argument is given returns true when none of the collection members are false or nil
  #my_all? when a Regex is passed as an argument returns false if any of the collection does not match the Regex
  #my_any? returns false if the block ever returns a value false or nil
  #my_any? returns false if the block ever returns a value false or nil::range
  #my_any? when no block or argument is given returns false if at least one of the collection is not true
 #my_none? when no block or argument is given returns false only if one of the collection members is true

andresporras3423 reviewed (changes requested):
### Status: Changes Required :recycle:
**Hello Shweta Srivastava** 
Good job with the improvements for this project :muscle:  :muscle:
Your methods meets passes most of the tests :clap: :clap: :clap:  
There are just a few details to improve before you can go ahead:

- [ ] Fix rubocop offenses
- [ ] Don't overuse rubocop disable
- [ ] #my_all? when no block or argument is given returns false when one of the collection members are false or nil
      `[true,[true],false].all? == [true,[true],false].my_all?`
- [ ] #my_any? returns false if the block ever returns a
value false or nil::range
    `(1..3).any?(&proc{|x| x==0}) == (1..3).my_any?(&proc{|x| x==0})`
- [ ] #my_any? when no block or argument is given returns false if at least one of the collection is not true
      `[false, 0].any? == [false, 0].my_any?`
- [ ] #my_none? returns false if the block ever returns true for all elements::range
    `(1..3).none?(&proc{|num| num%2==0}) == (1..3).my_none?(&proc{|num| num%2==0})`
- [ ] #my_none? when no block or argument is given returns true only if none of the collection members is true
      `[false, nil,false].none? == [false, nil,false].my_none?`
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

vidhishweta01 committed:
completed task with all checks

vidhishweta01 committed:
completed task with all checks

vidhishweta01 committed:
completed task with all checks

vidhishweta01 committed:
removed linter errors

vidhishweta01 committed:
removed linter errors

vidhishweta01 committed:
removed linter errors

vidhishweta01 commented:
#my_all? when no block or argument is given returns false when one of the collection members are false or nil
[true,[true],false].all? == [true,[true],false].my_all?
 #my_any? returns false if the block ever returns a
value false or nil::range
(1..3).any?(&proc{|x| x==0}) == (1..3).my_any?(&proc{|x| x==0})
 #my_any? when no block or argument is given returns false if at least one of the collection is not true
[false, 0].any? == [false, 0].my_any?
 #my_none? returns false if the block ever returns true for all elements::range
(1..3).none?(&proc{|num| num%2==0}) == (1..3).my_none?(&proc{|num| num%2==0})
 #my_none? when no block or argument is given returns true only if none of the collection members is true
[false, nil,false].none? == [false, nil,false].my_none?
![Screenshot](https://user-images.githubusercontent.com/73999611/108078022-2e0c6600-7093-11eb-81db-9ac7fe4956c0.png)

mimipeshy commented:
### STATUS: CHANGES REQUESTED 🔴 ♻️ 

Hello, @vidhishweta01 👋🏾 I must commend your hard work in this project, you should be proud of yourself for coming this far. However, 4 more tests are still failing as shown below

- [ ] enumerables #my_any? returns false if the block ever returns a value false or nil::range
     Failure/Error: expect(range.my_any?(&false_block)).to eq(range.any?(&false_block))
     
       expected: true
            got: false

- [ ]  enumerables #my_any? when no block or argument is given returns false if at least one of the collection is not true
     Failure/Error: expect(false_array.my_any?).to be false_array.any?
     
       expected false
            got true

- [ ] enumerables #my_none? returns true if the block never returns true for all elements
     Failure/Error: expect(array.my_none?(&true_block)).to eq(array.none?(&true_block))
     
       expected: true
            got: false

- [ ]  enumerables #my_none? when no block or argument is given returns false only if one of the collection members is true
     Failure/Error: expect(true_array.my_none?).to be false
     
       expected false
            got true

-  Let's improve the readme file in the following sections
- [ ] Do away with the paragraph below the screenshot image. You can just write a brief description something like `In this project, I attempted to implement some of the more popular methods in the Ruby enumerable method. These are:` then list the methods in bullet form eg 
`
-  #my_each

- #my_each_with_index`

That's all :smile: kindly submit another review when done.

_Feel free to reach out to me on slack if you have any questions or something is unclear, or just comment on this PR and tag my name and I will be glad to help_
Regards
Peris Ndanu

vidhishweta01 commented:
Now all the cases were running fine 
1.enumerables #my_any? returns false if the block ever returns a value false or nil::range
Failure/Error: expect(range.my_any?(&false_block)).to eq(range.any?(&false_block))

 expected: true
      got: false
 2. enumerables #my_any? when no block or argument is given returns false if at least one of the collection is not true
Failure/Error: expect(false_array.my_any?).to be false_array.any?

 expected false
      got true
3. enumerables #my_none? returns true if the block never returns true for all elements
Failure/Error: expect(array.my_none?(&true_block)).to eq(array.none?(&true_block))

 expected: true
      got: false
 4. enumerables #my_none? when no block or argument is given returns false only if one of the collection members is true
Failure/Error: expect(true_array.my_none?).to be false

 expected false
      got true

![Screenshot from 2021-02-16 21-05-37](https://user-images.githubusercontent.com/73999611/108085179-11742c00-709b-11eb-958a-06368c6d28c1.png)

vidhishweta01 committed:
corrected as per code review

BrianSammit commented on test5.rb:
```diff
@@ -0,0 +1,247 @@
+def multiply_els(array)
+  p array.my_inject(1) { |r, i| r * i }
+end
+rang = Range.new(5, 10)
+multiply_els(rang)
+multiply_els([23, 34, 56])
```
From line 240 to 242 is debugging code, you either can delete or keep it on another branch, [Here ](https://questions.microverse.org/t/debugging-instructions-code/346) you can read about why you shouldn't commit debugging code.

BrianSammit reviewed (changes requested):
CHANGES REQUIRED!! :recycle: :red_circle: 
--------------
Hello team!! :wave: you are working really hard on this project congrats :muscle: :1st_place_medal: 
There are a few changes to make to full fit this project :hammer_and_wrench: see them below :eyes: :point_down: 
-------

- [ ] Improve your Readme, it looks messy. Remember that the readme should look professional so the recruiters can see that you are professional and that is going to help you to land your first job. Use bullet points to improve it, write every method in a different line so it can be read easily. 
- [x] #my_any? when no block or argument is given returns false if at least one of the collection is not true. Instead is returning True.
- [x] #my_none? returns true if the block never returns true for all elements. instead is returning False.
- [x] #my_none? when no block or argument is given returns false only if one of the collection members is true. Instead is returning True.
- [ ] Improve your commit message. 
- You are often repeate the same message commit e.g `fixed linter errors`. Next time you can make the changes on linters and follow the step from [Here](https://docs.github.com/en/github/committing-changes-to-your-project/changing-a-commit-message) just follow the (Amending older or multiple commit messages) section. with that, you are creating a new commit that replaces the old one.
- Write your commit message in the imperative. [Here ](https://github.com/trein/dev-best-practices/wiki/Git-Commit-Best-Practices#formatting-rules) you can find Git Commit Best Practices. Read (Formatting Rules) section to write better commit.

- [ ] Don't commit debugging code. See my inline comments to know what I'm talking about :eyes: 
-------------
Make the required changes :hammer_and_wrench: :point_up: and send it for another review. :muscle: 
HAPPY CODING!! 

![AngelicConcreteHypsilophodon-max-1mb](https://user-images.githubusercontent.com/64173189/108101936-ec10ff00-7055-11eb-9490-141129c119e7.gif)

GzimAsani reviewed (changes requested):
### Status: Changes Required :recycle:
**Hello team** I know that you are trying so hard to finish this project but unfortunately I cannot accept it without the implementation of the following:
- [ ] Your rubocop is failing locally, currently you have 13 offenses
- [ ] Failures:
  1) enumerables #my_any? when no block or argument is given returns false if at least one of the collection is not true
  2) enumerables #my_none? returns true if the block never returns true for all elements
  3) enumerables #my_none? when no block or argument is given returns false only if one of the collection members is true

### **Please make the required changes and submit another review request when you are ready.**

<hr>
### if you have any question kindly reach me on slack @GzimAsani

### HAPPY CODING :computer:

vidhishweta01 commented:
hii my_any my_none are behaving the same way as the original methods I have run the cases there is a screenshot which shows the result 
![Screenshot from 2021-02-16 21-05-37](https://user-images.githubusercontent.com/73999611/108107462-47260e80-70b5-11eb-842b-ce29b2cddcd4.png)

vidhishweta01 commented:
Hii I have checked every case for it and my_any and my_none are behaving same as original method
[Screenshot from 2021-02-17 00-24-18](https://user-images.githubusercontent.com/73999611/108108347-86a12a80-70b6-11eb-89c3-78e228bb70d6.png)

vidhishweta01 committed:
corrected as per code review

vidhishweta01 committed:
finished with all the required changes

vidhishweta01 committed:
finished with all the required changes

vidhishweta01 commented:
All the below mentioned cases are tested and they are working same as original methods I have attached the result screenshot please go through it and If anything is required kindly elaborate with test cases.
1. enumerables #my_any? when no block or argument is given returns false if at least one of the collection is not true
2. enumerables #my_none? returns true if the block never returns true for all elements
3. enumerables #my_none? when no block or argument is given returns false only if one of the collection members is true
![Screenshot from 2021-02-17 00-24-18](https://user-images.githubusercontent.com/73999611/108110323-18119c00-70b9-11eb-85ec-c260324218ab.png)
![Screenshot from 2021-02-16 21-05-37](https://user-images.githubusercontent.com/73999611/108110371-2c559900-70b9-11eb-977a-8aa4055cf4b8.png)
![Screenshot](https://user-images.githubusercontent.com/73999611/108110394-35df0100-70b9-11eb-9010-ccb34097e3df.png)

GzimAsani reviewed (changes requested):
### Status: Changes Required :recycle:
**@vidhishweta01** Good job implementing the requirements but before we can move on you have to fix rubocop locally, currently you have 23 offenses [Screenshot](https://imgur.com/a/tJGasL5?fbclid=IwAR3krtNebZXIDNIQpM9P1ILy-VjpOrk8eXbqRKShe7wgp8-DtvnlTNrWO8M):
- [ ] Fix rubocop issues

### **Please make the required changes and submit another review request when you are ready.**

<hr>
### if you have any question kindly reach me on slack @GzimAsani

### HAPPY CODING :computer:

vidhishweta01 committed:
checked every case

vidhishweta01 committed:
checked every case

vidhishweta01 committed:
checked every case

vidhishweta01 committed:
checked every case

vidhishweta01 committed:
removed linters

vidhishweta01 committed:
changes made as per code review

vidhishweta01 committed:
final submission

vidhishweta01 committed:
final submission

vidhishweta01 committed:
final submission

vidhishweta01 committed:
fix linters

vidhishweta01 committed:
fix linters

vidhishweta01 commented:
All the below mentioned cases are tested and they are working same as original methods I have attached the result screenshot please go through it and If anything is required kindly elaborate with test cases.

enumerables #my_any? when no block or argument is given returns false if at least one of the collection is not true
enumerables #my_none? returns true if the block never returns true for all elements
enumerables #my_none? when no block or argument is given returns false only if one of the collection members is true
![Uploading Screenshot from 2021-02-17 01-03-48.png…]()

vidhishweta01 commented:
![Screenshot from 2021-02-17 00-24-18](https://user-images.githubusercontent.com/73999611/108162745-bcc0c780-7113-11eb-88f8-274476fb3296.png)
![Screenshot from 2021-02-16 21-05-37](https://user-images.githubusercontent.com/73999611/108162750-be8a8b00-7113-11eb-9898-ac23e8ae1a28.png)

arslanbisharat reviewed (changes requested):
### STATUS: CHANGES REQUIRED 🔴

Hey @vidhishweta01, Thanks for the great work done on the project. Your enumerable methods work just fine except for a few test cases.

- [ ] #my_any? when no block or argument is given returns false if at least one of the collection is not true

- [ ] #my_none? returns false if the block ever returns true for all elements::range

- [ ] #my_none? when no block or argument is given returns false only if one of the collection members is true

- [ ] #my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex

- [ ] #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

- [ ] #my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern

**Kindly make all the requested changes 💯 and submit another CRR. 👯**
Happy Coding! 😄

Reach me on slack for a ChitChat 🆘 : Arslan Bisharat

vidhishweta01 committed:
checked all the cases

vidhishweta01 committed:
checked all the cases

vidhishweta01 committed:
checked all the cases

vidhishweta01 committed:
fix linter errors

vidhishweta01 committed:
fix linter errors

vidhishweta01 commented:
all below cases are corrected
1. #my_any? when no block or argument is given returns false if at least one of the collection is not true

2. #my_none? returns false if the block ever returns true for all elements::range

3. #my_none? when no block or argument is given returns false only if one of the collection members is true

4. #my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex

5. #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

6. #my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern

mimipeshy commented:
### STATUS :APPROVED :partying_face: 

Hey there @vidhishweta01  wave: :wave:  good job on working on the requested changes. This project is therefore approved. Go ahead and merge. All the best in the next milestones :man_technologist: 

Regards
Peris Ndanu
